storage/azure/src/test/integration/backend/AzureServerStorage.test.ts

```diff
@@ -0,0 +1,112 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import "reflect-metadata";
+
+import { BlobClient, ContainerClient } from "@azure/storage-blob";
+
+import {
+  buildObjectKey,
+  buildObjectReference,
+} from "@itwin/object-storage-core/lib/common/internal";
+
+import { ObjectReference, TransferConfig } from "@itwin/object-storage-core";
+import {
+  InternalStorageIntegrationTests,
+  TestRemoteDirectory,
+  TestRemoteDirectoryManager,
+  testDeleteObjectWithTransferConfig,
+  testListObjectsWithTransferConfig,
+} from "@itwin/object-storage-tests-backend";
+
+import {
+  assertAzureTransferConfig,
+  buildBlobUrl,
+} from "../../../common/internal";
+import { AzureServerStorageBindings } from "../../../server";
+import { ServerStorageConfigProvider } from "../ServerStorageConfigProvider";
+
+describe("AzureServerStorage internal tests", () => {
+  const serverStorageConfig = new ServerStorageConfigProvider().get();
+  const config = {
+    ServerStorage: {
+      dependencyName: "azure",
+      instanceName: "primary",
+      ...serverStorageConfig,
+    },
+  };
+
+  const testInfrastructure = new InternalStorageIntegrationTests(
+    config,
+    AzureServerStorageBindings
+  );
+  const testDirectoryManager = new TestRemoteDirectoryManager(
+    testInfrastructure.serverStorage
+  );
+
+  beforeEach(async () => {
+    await testDirectoryManager.purgeCreatedDirectories();
+  });
+
+  after(async () => {
+    await testDirectoryManager.purgeCreatedDirectories();
+  });
+
+  it(`should delete object using transfer config`, async () => {
+    const deleteFunction = async (
+      reference: ObjectReference,
+      transferConfig: TransferConfig
+    ) => {
+      assertAzureTransferConfig(transferConfig);
+      const blobUrl = buildBlobUrl({
+        transferConfig,
+        reference,
+      });
+      const blobClient = new BlobClient(blobUrl);
```

Comment:
nitpick: it would be better if the function itself was not nested within the test. It is now a bit hard to read what the test does since the main function calls are at the bottom